arch/ia32/syscalls.S

```diff
@@ -32,27 +32,15 @@ sym: \
 .size sym, .-sym
 
 
-.bss
-vfork_address:
-.space 4
-.text
 .globl vfork
 .type vfork, @function
 .align 4, 0x90
 vfork:
-	mov (%esp), %eax
-	mov %eax, (vfork_address)
+	/* Save the return address as child will override parent's stack. */
+	mov (%esp), %edx
```

Comment:
Yes, they can be destroyed by callee during normal call, but as far as I know `edx` is saved in `interrupts_pushContext` and later restored by `interrupts_popContextUnlocked`.

Reply:
Ok, also, now I see that it has to stay that way, otherwise we would need to save callee-save register ourself for user app sake and we're back in square one :( 

Perhaps a comment that we're depending on kernel saving all registers regardless of ABI would be helpful

Reply:
I added deatailed commend